### ISSUE TITLE
PyCaffe MemoryError loading model

### ISSUE THREAD

**Issue by SimoV8** [BVLC/caffe#3042]
Hi, I'm trying to load the VGG 16-layer model (downloaded from here http://www.robots.ox.ac.uk/~vgg/research/very_deep/) with pycaffe but I always get a MemoryError message when I try to execute `net = caffe.Net(prototxt, model, caffe.TEST)`. I'm running it on a Jetson TK1 board that has about 2GB of ram and more then 1GB of memory is free to use. I changed the prototxt file reducing the first input dimension from 10 to 1 and I removed the last 3 layers , cutting the network at the relu7 layer, to reduce memory usage. 

When I run the script debug messages say "Memory required for data: 114583552" and then "The total number of bytes read was 553432081". 
Summing up these numbers, it seems that the memory allocation should be around 668 MB. Considering an expansion of the model in memory and other variables, it shouldn't exceed 1GB, that fits with my hw requirements. The problem is that probably I'm missing something because when I run the script it takes much more then 1GB and when it finish all the available memory it crashes. Can someone explain me why? How can I solve that?
I also tried to use the modified version of caffe provided by the #2009 pull request but without success.

PS: I tried both with gpu and cpu mode. The cuda version installed is 6.5 with cuDNN v2.

**Comment by seanbell** [comment]
That reported number is just for blob storage _between_ the layers.  It does not include all of the internal buffers _inside_ each layer (e.g. inside each convolution layer).  As far as I know there's currently no code for counting up all of the internal buffers, as they're hidden inside of the layer classes.

**Comment by longjon** [comment]
Closing as a usage issue; discuss on caffe-users as desired.

From https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md:
> _Please do not post usage, installation, or modeling questions, or other requests for help to Issues._
Use the [caffe-users list](https://groups.google.com/forum/#!forum/caffe-users) instead. This helps developers maintain a clear, uncluttered, and efficient view of the state of Caffe.
